### PR TITLE
automatically compute max_step for LPA in boost

### PR DESCRIPTION
**Pull request by MaxThevenet**
`max_step` can be automatically computed so that the simulation stops when the back of the simulation domain reaches location `zmax_plasma_to_compute_max_step`.

`zmax_plasma_to_compute_max_step` is an optional input parameter: typically set it to the end of the plasma. When specified, `max_step` is overwritten.

### CODE REVIEW

**Comment by MaxThevenet** [comment]
@RemiLehe do you think we can merge this PR?

File: Docs/source/running_cpp/parameters.rst [hunk 1]
```diff
@@ -28,6 +28,15 @@ Overall simulation parameters
     The direction of the Lorentz-transform for boosted-frame simulations
     (The direction ``y`` cannot be used in 2D simulations.)
 
+* ``warpx.zmax_plasma_to_compute_max_step`` (`float`) optional (default `-inf`)
+    Used when running in a boosted frame. If specified, automatically
```
Comment (RemiLehe) [inline Docs/source/running_cpp/parameters.rst]: Could you remove `(default -inf)`?
Also should we replace `Used when running...` by `Can be useful when running...`, just to emphasize that this is optional?

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -272,6 +274,10 @@ WarpX::ReadParameters ()
 
     ReadBoostedFrameParameters(gamma_boost, beta_boost, boost_direction);
 
+    do_compute_max_step_from_zmax = 
+        pp.query("zmax_plasma_to_compute_max_step", 
+                  zmax_plasma_to_compute_max_step);
```
Comment (RemiLehe) [inline Source/WarpX.cpp]: Could you add a quick comment explaining that `pp.query` returns 1 if  `zmax_plasma_to_compute_max_step` is in the input deck?

File: Docs/source/running_cpp/parameters.rst (hunk 1 above)
Comment (MaxThevenet) [reply Docs/source/running_cpp/parameters.rst]: Sure, this was included in the latest commit

File: Source/WarpX.cpp (hunk 2 above)
Comment (MaxThevenet) [reply Source/WarpX.cpp]: Sure, this was included in the latest commit

**Comment by MaxThevenet** [comment]
@RemiLehe Thanks for your comments!

**Pull request merged**